### DISCUSSION TITLE
Emulate Eclipse Run Configuration

### DISCUSSION THREAD

**Discussion by lppedd** [eclipse-tycho/tycho#1302] [Q&A]
In an effort to minimize the interactions with the Eclipse UI (and move away from it), I'm converting some manual procedures done through the Eclipse UI to Maven goals. An example is the generation of EMF definitions using the `eclipse-run` plugin.

Now, is it possible to do the same for the Run Configuration?
My environment is made up of multiple plugin bundles, which target Eclipse 4.23.

**Comment by laeubi** [comment]
> In an effort to minimize the interactions with the Eclipse UI (and move away from it), I'm converting some manual procedures done through the Eclipse UI to Maven goals. An example is the generation of EMF definitions using the `eclipse-run` plugin.
> 
> Now, is it possible to do the same for the Run Configuration? My environment is made up of multiple plugin bundles, which target Eclipse 4.23.

If you convert your "run config" into a product you can use `eclipse-run` plugin as well.

**Reply by lppedd** [reply]
Thanks! The point is I'm not building a standalone RCP application. I'm just extending Eclipse with new plugins.
Does this apply anyway?

**Reply by laeubi** [reply]
Would it then not make more sense to run from Eclipse directly? Anyways "Eclipse" is also build by Tycho using product files, so you should be able to create one as well.

**Reply by lppedd** [reply]
@laeubi I'm now emulating the Run Configuration by building the same command line string that Eclipse itself generates (you can see it via the `Show command line` button in the Run Configuration dialog).
A bit more work but it seems to be working.
